tests/tasty/Arbitrary.hs

```diff
@@ -155,3 +174,109 @@ instance Arbitrary Constant where
                     , L <$> arbitrary <*> arbitrary
                     ]
   shrink = genericShrink
+
+-- | Used in UndoANFTests.
+newtype AnfSymbol = AnfSymbol { unAnfSymbol :: Symbol }
+  deriving (Eq, Show, Generic)
+instance Arbitrary AnfSymbol where
+  arbitrary = AnfSymbol . mappendSym anfPrefix <$> arbitrary
+
+-- | This instance does **not** create Refts with anf symbols.
+instance Arbitrary Reft where
+  arbitrary = reft <$> arbitrary <*> arbitrary
+
+-- | This instance does **not** create SortedRefts with anf symbols.
+instance Arbitrary SortedReft where
+  arbitrary = arbitrarySortedReft (const arbitrary) 1
+
+arbitrarySortedReft :: (Int -> Gen Symbol) -> Int -> Gen SortedReft
+arbitrarySortedReft symGen = \i -> do
+  sort <- arbitrary
+  eq <- oneof $ pure <$> [Eq, Ueq]
+  sym <- symGen i
+  expr <- arbitrary
+  pure $ RR sort $ reft sym (PAtom eq (EVar sym) expr)
+
+-- | Base environment with no declared properties; do not add an Arbitrary
+-- instance to this and instead use newtypes.
+newtype Env = Env { unEnv :: M.HashMap Symbol SortedReft }
+  deriving (Eq, Show, Generic)
+
+-- | Env without anf vars.
+newtype NoAnfEnv = NoAnfEnv { unNoAnfEnv :: Env }
+  deriving (Eq, Show, Generic)
+instance Arbitrary NoAnfEnv where
+  arbitrary = NoAnfEnv <$> (arbitraryEnv 4 gen)
```

Comment:
Shouldn't the size be taken from the monad, so the caller can decide how large an environment to create? Also, the size here is irrelevant as gen always produces a single pair. Shouldn't we test with larger environments?

Reply:
Changed.

Reply:
Sorry, I changed it for FlatAnfEnvs, I think.  I have now changed it for `NoAnfEnv` too.